### PR TITLE
Initial classes

### PR DESCRIPTION
**Pull request by mecha**

### CODE REVIEW

File: src/Exception/AbstractBaseTransitionerException.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+
+/**
+ * Abstract base functionality for exceptions thrown in relation to a transitioner.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractBaseTransitionerException extends RootException implements TransitionerExceptionInterface
```
Comment (XedinUnknown) [inline src/Exception/AbstractBaseTransitionerException.php]: Consider extending [`AbstractBaseException`].

[`AbstractBaseException`]: https://github.com/Dhii/exception/blob/develop/src/AbstractBaseException.php#L7

File: src/Exception/TransitionerException.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+class TransitionerException extends AbstractBaseTransitionerException
+{
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null     $message      The error message, if any.
+     * @param int|null                   $code         The error code, if any.
```
Comment (XedinUnknown) [inline src/Exception/TransitionerException.php]: The [`InitBaseExceptionCapableTrait`][] should probably be used as a standard for what arguments and types an exception should accept.

[`InitBaseExceptionCapableTrait`]: https://github.com/Dhii/exception/blob/develop/src/InitBaseExceptionCapableTrait.php#L27

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/Exception/TransitionerException.php (hunk 2 above)
Comment (mecha) [reply src/Exception/TransitionerException.php]: [`InitBaseExceptionCapableTrait`](https://github.com/Dhii/exception/blob/develop/src/InitBaseExceptionCapableTrait.php#L27) is part of [`ExceptionTrait`](https://github.com/Dhii/exception/blob/develop/src/ExceptionTrait.php), which is `use`d by [`AbstractBaseException`](https://github.com/Dhii/exception/blob/develop/src/AbstractBaseException.php#L7). The abstract base exception in this package now extends that, so it's being used indirectly through inheritance.

Comment (XedinUnknown) [reply src/Exception/TransitionerException.php]: That's true. And yet, the documented types haven't changed.

File: src/Exception/CouldNotTransitionException.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    /**
+     * Sets the transition.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null $transition The transition, if any.
+     *
+     * @throws InvalidArgumentException If the argument is not a string or stringable object.
+     */
+    protected function _setTransition($transition)
```
Comment (XedinUnknown) [inline src/Exception/CouldNotTransitionException.php]: Looks like awareness can very easily be extracted into traits and put into [dhii/data-state-abstract][].

[dhii/data-state-abstract]: https://github.com/Dhii/data-state-abstract

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/Exception/TransitionerException.php (hunk 2 above)
Comment (mecha) [reply src/Exception/TransitionerException.php]: My bad. I didn't interpret your first message as meaning to use the same docs as that trait, but merely to `use` the it. Will correct.

**Review by XedinUnknown — APPROVED** [review]

**Pull request merged**